bridge/slack/slack.go

```diff
@@ -315,6 +321,39 @@ func (b *Bslack) sendRTM(msg config.Message) (string, error) {
 	return b.postMessage(&msg, messageParameters, channelInfo)
 }
 
+func (b *Bslack) updateTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (handled bool, err error) {
```

Comment:
Please don't use named returns unless you absolute have to. Especially given that you combine them with only standard returns below.

They should normally only be used for very short functions as they tend to make a function's logic much less readable.

Reply:
Is it possible to have named returns but still explicitly return values? Seems to be perfectly fine as documentation, no? https://tour.golang.org/basics/7

Reply:
It is fine indeed syntax-wise. The core issue with named return variables is not the naming itself. It is that they implicitly allow for "naked returns" and those are the devils' own offspring and the source of many a production bug in systems ([the discussion here is a good read](https://github.com/golang/go/issues/21291)).

Given that the overall code of Matterbridge does not really have policy of using named returns I would at the moment suggest to not start introducing them randomly. We can potentially revisit this at a later point.

Reply:
Yep, I def prefer them as inline documentation as I am sometimes confused whats coming back when skimming, but def happy to defer that convo :) will remove when off mobile!

Reply:
Also, yay for this! https://github.com/alexkohler/nakedret

Reply:
Like @Helcaraxan I'm not a fan of named returns because of exactly the same reason